test/functional/p2p_blockfilters.py

```diff
@@ -48,11 +48,19 @@ def set_test_params(self):
         self.rpc_timeout = 480
         self.num_nodes = 2
         self.extra_args = [
-            ["-blockfilterindex", "-peerblockfilters"],
-            ["-blockfilterindex"],
+            ["-blockfilterindex=basic", "-peerblockfilters"],
+            ["-blockfilterindex=basic", "-peerblockfilters=0"],
         ]
 
     def run_test(self):
+        self.log.info('Test -peerblockfilters without -blockfilterindex raises init error')
+        err_msg = 'Error: Cannot set -peerblockfilters without -blockfilterindex.'
+        self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=0', '-peerblockfilters'], err_msg)
+
+        self.log.info('Test passing unknown -blockfilterindex type raises init error')
+        err_msg = 'Error: Unknown -blockfilterindex value foo.'
+        self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=foo', '-peerblockfilters'], err_msg)
```

Comment:
Just confirming, is it OK to call `assert_start_raises_init_error` when the node is already started?

For example, if I pass a valid `blockfilterindex` instead,

    self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=basic', '-peerblockfilters'], err_msg)

I thought I would get "test failed because node started successfully".

But instead, I get "test failed because expected error didn't match":

    AssertionError: [node 0] Expected message "Error: Cannot set -peerblockfilters without -blockfilterindex." does not fully match stderr:
    "Error: Cannot obtain a lock on data directory /var/folders/2b/58wl974967710d36y5sb2_440000gn/T/bitcoin_func_test_zz7jcluz/node0/regtest. Bitcoin Core is probably already running."

Reply:
I think it's ok because (a) it's the same as what I see when testing these manually, (b) this also tests that the args are checked first, and (c) it avoids needlessly stopping and restarting the node. Per `test/functional/README.md`, *Avoid stop-starting the nodes multiple times during the test if possible. A stop-start takes several seconds, so doing it several times blows up the runtime of the test.*